xlive/Blam/Engine/simulation/simulation_world.cpp

```diff
@@ -144,25 +133,67 @@ void c_simulation_world::apply_simulation_queue(const c_simulation_queue* queue)
 			element = queue->get_next_element(element);
 		}
 	}
+}
 
-	if (shell_time_now_msec() - last_time_msec > 1000)
+void c_simulation_world::destroy_update()
+{
+	for (int32 i = 0; i < k_simulation_queue_count; i++)
 	{
-		last_time_msec = shell_time_now_msec();
+		queue_get((e_simulation_queue_type)i)->clear();
 	}
 }
 
-void c_simulation_world::destroy_update()
+typedef void(__thiscall* t_c_simulation_world__initialize_world)(c_simulation_world*, int32, int32, int32);
+t_c_simulation_world__initialize_world p_c_simulation_world__initialize_world;
+
+void c_simulation_world::initialize_world(int32 a2, int32 a3, int32 a4)
 {
-	for (int32 i = 0; i < k_simulation_queue_count; i++)
+	p_c_simulation_world__initialize_world(this, a2, a3, a4);
+	if (!is_playback())
+	{
+		queues_initialize();
+	}
+}
+
+void __declspec(naked) jmp_initialize_world() { __asm { jmp c_simulation_world::initialize_world } }
+
+typedef void(__thiscall* t_c_simulation_world__destroy_world)(c_simulation_world*);
+t_c_simulation_world__destroy_world p_c_simulation_world__destroy_world;
+
+void c_simulation_world::destroy_world()
+{
+	// call orig
+	p_c_simulation_world__destroy_world(this);
+
+	// clear the queues
+	if (!is_playback())
 	{
-		c_simulation_queue::dispose(queue_get((e_simulation_queue_type)i));
+		queues_dispose();
 	}
 }
 
+void __declspec(naked) jmp_destroy_world() { __asm{ jmp c_simulation_world::destroy_world } }
+
 void c_simulation_world::queues_initialize()
 {
 	for (int32 i = 0; i < k_simulation_queue_count; i++)
 	{
 		queue_get((e_simulation_queue_type)i)->initialize();
 	}
 }
+
+void c_simulation_world::queues_dispose()
+{
+	for (int32 i = 0; i < k_simulation_queue_count; i++)
+	{
+		queue_get((e_simulation_queue_type)i)->dispose();
+	}
+}
+
+void simulation_world_apply_patches()
+{
+	// ### TODO dedi offset
+	DETOUR_ATTACH(p_c_simulation_world__initialize_world, Memory::GetAddress<t_c_simulation_world__initialize_world>(0x1DDB4E, 0x0), jmp_initialize_world);
+	DETOUR_ATTACH(p_c_simulation_world__destroy_world, Memory::GetAddress<t_c_simulation_world__destroy_world>(0x1DE0A9, 0x0), jmp_destroy_world);
```

Comment:
Offsets for server:
jmp_initialize_world: 0x1C500E
jmp_destroy_world: 0x1C5569

Reply:
thanks